src/instructlab/model/download.py

```diff
@@ -1,31 +1,258 @@
+                self.download_gguf()
+
+        except Exception as exc:
+            click.secho(
+                f"Downloading model failed with the following Hugging Face Hub error: {exc}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+
+    def download_gguf(self) -> None:
+        try:
+            hf_hub_download(
+                token=self.hf_token,
+                repo_id=self.repository,
+                revision=self.release,
+                filename=self.filename,
+                local_dir=self.download_dest,
+            )
+
+        except Exception as exc:
+            click.secho(
+                f"Downloading GGUF model failed with the following Hugging Face  Hub error: {exc}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+
+    def download_safetensors(self) -> None:
+        try:
+            os.makedirs(
+                name=os.path.join(self.download_dest, self.repository),
+                exist_ok=True,
+            )
+
+            snapshot_download(
+                token=self.hf_token,
+                repo_id=self.repository,
+                revision=self.release,
+                local_dir=os.path.join(self.download_dest, self.repository),
+            )
+        except Exception as exc:
+            click.secho(
+                f"Downloading safetensors model failed with the following Hugging Face  Hub error: {exc}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+
+
+class OCIDownloader(Downloader):
+    """
+    Class to handle downloading safetensors models from OCI Registries
+    We are leveraging OCI v1.1 for this functionality
+    """
+
+    def __init__(self, repository: str, release: str, download_dest: str, ctx) -> None:
+        super().__init__(
+            repository=repository, release=release, download_dest=download_dest
+        )
+        self.ctx = ctx
+
+    def _build_oci_model_file_map(self, oci_model_path: str) -> dict:
+        """
+        Helper function to build a mapping between blob files and what they represent
+        Format for the index.json file can be found here: https://github.com/opencontainers/image-spec/blob/main/image-layout.md#indexjson-file
+        """
+        index_hash = ""
+        index_ref_path = f"{oci_model_path}/index.json"
+        try:
+            index_ref = _load_json(Path(index_ref_path))
+            match = None
+            for manifest in index_ref["manifests"]:
+                if (
+                    manifest["mediaType"]
+                    == "application/vnd.oci.image.manifest.v1+json"
+                ):
+                    match = _extract_SHA(manifest["digest"])
+
+            if match:
+                index_hash = match.group(1)
+            else:
+                raise ValueError(
+                    f"could not find hash for index file at: {oci_model_path}"
+                )
+        except Exception as exc:
```

Comment:
This except block is redundant since it just re-raises the exception back. You can remove it and `try` and this will also un-indent the code.

---

That said, I wonder if instead of that, you'd want to reraise ValueError?

---

Same comments below (line 173).